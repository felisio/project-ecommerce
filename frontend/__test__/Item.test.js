import Item from '../components/Item'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const mockData = {
  id: 'Abc123',
  title: 'A cool item',
  price: 5000,
  description: 'Cool item desc',
  image: 'dog.js',
  largeImage: 'large-dog.js',
}

describe('<Item/>', () => {
  it('renders and display properly', () => {
    const wrapper = shallow(<Item item={mockData} />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders and display properly', () => {
    const wrapper = shallow(<Item item={mockData} />);
    // console.log(wrapper.debug())
    const PriceTag = wrapper.find('PriceTag')
    expect(PriceTag.children().text()).toBe('$50');
    expect(wrapper.find('Title a').text()).toBe(mockData.title)
    expect(wrapper.find('p').text()).toBe(mockData.description)
  });

  it('renders the img properly', () => {
    const wrapper = shallow(<Item item={mockData} />);
    expect(wrapper.find('img').props().src).toBe(mockData.image)
    expect(wrapper.find('img').props().alt).toBe(mockData.title)
  });

  it('renders the buttons properly', () => {
    const wrapper = shallow(<Item item={mockData} />);
    const buttonList = wrapper.find('.buttonList')
    expect(buttonList.children()).toHaveLength(3)
    expect(buttonList.find('Link')).toBeTruthy()
    expect(buttonList.find('AddToCart').exists()).toBe(true)
    expect(buttonList.find('DeleteItem')).toBeTruthy()
  });
});