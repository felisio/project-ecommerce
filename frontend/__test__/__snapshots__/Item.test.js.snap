// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and display properly 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.js"
    width="200"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "Abc123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    Cool item desc
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "Abc123",
          },
        }
      }
    >
      <a>
         Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="Abc123"
    />
    <DeleteItem
      id="Abc123"
    >
      Delete this item 🗑️
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
